unlock-js/src/__tests__/web3Service.test.js

```diff
@@ -636,6 +483,92 @@ describe('Web3Service', () => {
     })
   })
 
+  describe('inputsHandlers', () => {
```

Comment:
I understand the limited nature of this approach, testing only that the function takes inputs and puts out the correct values. The tests used to be this way, but I changed them because we had a subtle bug where the way the inputs were parsed changed, and the tests didn't catch it. Testing against the actual function inputs is a necessary test.

Could we add back in some tests, perhaps version-specific, that verify our input parsing is the same as we expect? Sort of a mini-integration test

Reply:
These broken tests are preventing an upgrade to v1 which is the highest priority.

Reply:
To be honest I am not even sure what these tests are for.

Reply:
how do the tests break?

Reply:
the tests for `_parseTransactionFromInput` don't actually verify that the parsed inputs from real events are parsed correctly and passed to inputsHandlers in a format it expects. These tests verify that. When I was moving from web3 to ethers, the tests all passed, even though web3Service was broken. This was a case of mocking out too much.

I'm fine with opening a new issue to add these tests back. Can you describe how they were failing?

Reply:
They are version specific which means that, at best, they test a single version... so not sure they actually fulfill their role.